src/core/server/plugins/discovery/plugins_discovery.ts

```diff
@@ -46,9 +46,9 @@ export function discover(config: PluginsConfig, coreContext: CoreContext) {
   const log = coreContext.logger.get('plugins-discovery');
   log.debug('Discovering plugins...');
 
-  if (config.additionalPluginPaths.length) {
+  if (config.additionalPluginPaths.length && coreContext.env.mode.prod) {
     log.warn(
-      `Explicit plugin paths [${config.additionalPluginPaths}] are only supported in development. Relative imports will not work in production.`
+      `Explicit plugin paths [${config.additionalPluginPaths}] should only be used in development. Relative imports may not work properly in production.`
```

Comment:
Not sure if we should be **even** mode explicit by explaining that relative path used for the plugin should be the same in dev than in prod. But the root issue should be addressed by https://github.com/elastic/kibana/issues/40446 anyway.

Reply:
I feel like this warning in production is pretty useless. If the plugin is working with `--plugin-path` this warning doesn't tell you anything useful that an admin can or should do.

I think we should either remove this completely or have a different warning only in dev that warns about relative paths will break if the plugin is installed in a different location in production. Though that warning would also be less relevant after #40446.

Reply:
You are right, logging the warning in dev probably makes more sense than in production